### PR TITLE
Move ARCH_HAVE_NESTED_INTERRUPT macro under CORTEXM7

### PR DESCRIPTION
**Pull request by drashti304**
Move ARCH_HAVE_NESTED_INTERRUPT macro under CORTEXM7, as nested interrupt feature  is implemented and tested on CORTEXM7. 
Not all other architecture supports it.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cddb451dacd2c109efb64d038cba417a3be9677f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cddb451dacd2c109efb64d038cba417a3be9677f] - Code Rule Check (C++) OK.

File: build/configs/qemu/tc_64k/defconfig [hunk 1]
```diff
@@ -246,7 +246,6 @@ CONFIG_BOARD_LOOPSPERMSEC=4531
 #
 CONFIG_ARCH_HAVE_INTERRUPTSTACK=y
 CONFIG_ARCH_INTERRUPTSTACK=0
-CONFIG_ARCH_HAVE_NESTED_INTERRUPT=y
```
Comment (sijupunnnoose) [inline build/configs/qemu/tc_64k/defconfig]: why you removed this??

**Comment by sunghan-chang** [comment]
@drashti304 duplicated with #3500

**Comment by sunghan-chang** [comment]
@drashti304 Let's close this because this is duplicated

**Comment by seinfra** [comment]
Target : [096501f38e16cbbfc369df6b3f503ea1f14b1b79] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [096501f38e16cbbfc369df6b3f503ea1f14b1b79] - Code Rule Check (C++) OK.

**Pull request merged**